src/adapters/MeetingsJSONAdapter.js

```diff
@@ -143,13 +157,25 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
       /* eslint-enable no-confusing-arrow */
     );
 
-    // Send updates on the meeting when a mute event is triggered
-    const muteEvent$ = fromEvent(document, MUTE_AUDIO_CONTROL).pipe(
+    // Send updates on the meeting when an action is triggered
+    const audioEvents$ = fromEvent(document, MUTE_AUDIO_CONTROL);
+    const videoEvents$ = fromEvent(document, MUTE_VIDEO_CONTROL);
+    const joinEvents$ = fromEvent(document, JOIN_CONTROL);
+    const leaveEvents$ = fromEvent(document, LEAVE_CONTROL);
+
+    const events$ = merge(audioEvents$, videoEvents$, joinEvents$, leaveEvents$).pipe(
       filter((event) => event.detail.ID === ID),
       map((event) => event.detail)
     );
 
-    return concat(getMeetingWithMedia$, muteEvent$);
+    return concat(getMeetingWithMedia$, events$).pipe(
+      map((meeting) => {
```

Comment:
This could be a `tap`
```suggestion
      tap((meeting) => this.datasource[ID] = meeting)
```